stjude/proteinpaint
# use Mds3 type to catch potential typescript errors early

## Description

in `termdb/config` route handler functions. NOTE: Line 156 of `termdb.config.ts` replicates a previous error, would need to be deleted before approving this PR. Also, need to address other type check errors.

@creilly8 I deleted the `hicdata.genomes.ts` route, please verify that's okay

To test: 
- `cd proteinpaint/server; npx tsc` should emit an error like`// routes/termdb.config.ts:156:6 - error TS18048: 'ds.queries' is possibly 'undefined'.`
- the unit test CI should also fail because it calls `npx tsc` in one of the workflow job steps (see CI results of the first commit to this branch)

CI results of first branch commit (NOTE: already fixed by subsequent commits): There were a bunch of type errors detected based on potentially undefined properties like `termdb.cohort` in the `Mds3` type. Either use the `?.` optional chaining syntax in the functions, or update the Mds3 termdb.cohort to always require a `termdb.cohort` property (maybe not the case)? 


## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
